### PR TITLE
Migrate Calibration/TkAlCaRecoProducers to esConsumes and other fixes

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The main goal of this PR is to migrate to `esConsumes` the `Calibration/TkAlCaRecoProducer` subsystem (as part of issue https://github.com/cms-sw/cmssw/issues/31061)
I profit of this PR to perform also the following:

   * remove few strays `#include "FWCore/Framework/interface/ESHandle.h"` left from previous migrations
   * address the `"UnkownType"` => `"UnknownType"` typo highlighted in https://github.com/cms-sw/cmssw/pull/32363#discussion_r537865648
   * `Calibration/TkAlCaRecoProducers/plugins/PCLMetadataWriter.h` is removed and all code included in the `.cc` (as recommended for plugins).

#### PR validation:

`cmssw` compiles.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, no backport needed.

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc [hunk 1]
```diff
@@ -1173,9 +1173,9 @@ bool PrimaryVertexValidation::isHit2D(const TrackingRecHit& hit, const PVValHelp
         else if (dynamic_cast<const ProjectedSiStripRecHit2D*>(&hit))
           return false;  // crazy hit...
         else {
-          edm::LogError("UnkownType") << "@SUB=PrimaryVertexValidation::isHit2D"
-                                      << "Tracker hit not in pixel and neither SiStripRecHit2D nor "
-                                      << "SiStripMatchedRecHit2D nor ProjectedSiStripRecHit2D.";
+          edm::LogError("UnknownType") << "@SUB=AlignmentTrackSelector::isHit2D"
```
Comment (christopheralanwest) [inline Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc:1176]: Why has this been modified from `PrimaryVertexValidation::isHit2D`? At https://github.com/cms-sw/cmssw/pull/32432/files#diff-2b4e53d29d0035ce6ff0c0f3ca63fcc8c860fcd91a70bd68bdf4e1fd491031cbR2574

File: Alignment/APEEstimation/plugins/ApeEstimator.cc [hunk 2]
```diff
@@ -2571,9 +2571,9 @@ bool ApeEstimator::isHit2D(const TrackingRecHit& hit) const {
           const ProjectedSiStripRecHit2D* pH = static_cast<const ProjectedSiStripRecHit2D*>(&hit);
           return (this->isHit2D(pH->originalHit()));  // depends on original...
         } else {
-          edm::LogError("UnkownType") << "@SUB=AlignmentTrackSelector::isHit2D"
-                                      << "Tracker hit not in pixel, neither SiStripRecHit[12]D nor "
-                                      << "SiStripMatchedRecHit2D nor ProjectedSiStripRecHit2D.";
+          edm::LogError("UnknownType") << "@SUB=AlignmentTrackSelector::isHit2D"
```
Comment (christopheralanwest) [inline Alignment/APEEstimation/plugins/ApeEstimator.cc:2574]: Should this read `ApeEstimator::isHit2D` instead?

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@christopheralanwest ping

**Comment by christopheralanwest** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**